olm/operator-registry.md

```diff
@@ -0,0 +1,322 @@
+spec:
+  containers:
+  - name: operator-manifest
+    image: quay.io/community-operators/foo@sha256:abc123
+    command: ['operator-registry', 'serve' '-manifestdir "./manifests/"']
+  initContainers:
+  - name: init-manifest
+    image: quay.io/operator-framework/init-operator-manifest:latest
+    command: ['cp', 'operator-registry', '.']
+```
+
+We will also need to update operator-registry to include a new `serve` command that knows how to parse the manifest/ folder in the operator bundle image. This command will write to a configmap that OLM will use to get the bundle data.
+
+*DB Versioning*
+
+Since we are now calling this database schema a first class releasable object, we also need to be able to make changes to the schema over time. Along with that, we will need to be able to migrate to newer schema versions over time. To get started with that, we will update the existing operator-registry database to include a table that defines the current version of the database schema. At that point, the DB format is versioned with schemas.
+
+To support this, we will also add a version to the built operator-registry binary tooling (which currently exists in an unversioned state). Going forward, this tooling will support a set of different versions and handle migrations as needed to the latest schema that the tool's version supports. To go along with this, the runtime GRPC API that serves data to OLM will be versioned with the rest of the tooling (by the nature of the space it exists in) but will remain backwards compatible for API calls.
+
+Any large changes we wish to make (i.e. want to deprecate the current grpc api because we've changed how dependency resolution works) will be indicated via new CatalogSource types (current types are grpc, and configmap/internal). We will also provide a migration strategy along with any such changes.
+
+*Make CatalogSource type for this image*
+
+Finally, once these operator registry images are servable, we will update the catalog source to be able to download these images and serve them.
+
+Additionally, we will update the spec to create a polling field that allows the index to be updated automatically on a configurable poll timer.
+
+```
+apiVersion: operators.coreos.com/v1alpha1
+kind: CatalogSource
+metadata:
+  name: example-catalog
+  namespace: olm
+spec:
+  sourceType: grpc
+  image: quay.io/operator-framework/community-operators:
+  poll:
+    interval: 1m
+```
+
+Some open questions about the actual poll implementation: In openshift, for many of the the same reasons that OLM cannot directly pull the operator bundle image, OLM cannot query for new shas on a given tag directly. In order to accomplish this, we will need to create a deployment process that has an implicit understanding of what images are available (for example, we may use a rollout to force the deployment to update on the timer -- with a pull policy of always this will result in a new pod that is on the latest image when there are updates).
```

Comment:
Can we please write down how we are going to solve this?

Reply:
Updated the doc to specify.